### DISCUSSION TITLE
The Library class

### DISCUSSION THREAD

**Discussion by balhoofd** [beetbox/beets#5082] [General]
Hello,

I'm tinkering a bit with writing a scraper for my movies and series and needed a database solution. I figured since I already use Beets, I might as well use the functionality you already written to access a database. I've checked the docs for devs but here's what I don't really understand or I'm looking at it wrong and was hoping to get some clarification here.

My purpose is to have a database of my movies and series and `Library` looks like what I need. Now, from the docs it would seems I should start with `Item`

`Each Item object represents a song or track. (We use the more generic term item because, one day, beets might support non-music media.)`

A movie or episode is a non-music media. 

But when I check the source code for `Item` it says `Represent a song or track.` and has a bunch of fields related to songs. So inheriting seems wierd to make a `Movie` or `Episode` class and I feel I should inherit `LibModel`

Is `Item` an unfortunate chosen name or am I wrong in my assessment?

**Comment by Serene-Arc** [comment]
Hi! It says 'one day' because we currently don't support non-music media. Beets is set up to handle music, and the library class is built around that assumption, as well as around Musicbrainz data and formats. Because of that, it might take a considerable amount of effort to be able to use the beets code for movies and TV shows.

Beets is based on sqlite though; it might be simpler to use a library that gets data from IMDB and write your own database handler that is as simple or as complicated as you need.

**Comment by balhoofd** [comment]
I have written a scraper that works similar to the importer of Beets, I'm looking for solutions to store that data. Beets has already a handler that can communicate with SQL and that's the part I don't want to write.

I mean, I could in theory just use `dbcore` and write my own `Library` but I was checking if I could use existing code. Hence why I came across `Item`.

Yes, indeed it says 'one day' but it threw me off since `Item` has now all these fields for songs and then 'one day' one would need to rename `Item` or make a new `Song` class with every song specific field transferred to `Song`. This seems like a lot of trouble than just naming `Item` `Song` from the beginnen.

Again, I might be looking at this the wrong way.

**Reply by Serene-Arc** [reply]
Sorry, I'm not sure what to tell you. You can look up the person who wrote that comment and ask them if you'd like, but afaik there are no plans to expand the purview of beets at the moment.

**Reply by balhoofd** [reply]
I know. Just figuring out how to begin with my little project.

Guess I'll just inherit LibModel and see from there. 

Thanks for taking an interest.
